### PR TITLE
Ignore not existing logs when collecting log files

### PR DESCRIPTION
**Pull request by kokosing**
Ignore not existing logs when collecting log files

Fixes https://github.com/trinodb/trino/issues/7577

### CODE REVIEW

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/DockerContainer.java [hunk 1]
```diff
@@ -350,7 +350,7 @@ private void copyFileFromContainer(String filename, Path targetPath)
         try {
             return Splitter.on("\n")
                     .omitEmptyStrings()
-                    .splitToList(execCommand("/usr/bin/find", path, "-type", "f", "-print"));
+                    .splitToList(execCommand("test", "-d", path, "&&", "/usr/bin/find", path, "-type", "f", "-print"));
```
Comment (findepi) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/DockerContainer.java:353]: is it actually executed with a shell?

Comment (findepi) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/DockerContainer.java:353]: indeed, it is. The use of `String[]` is misleading here (`io.trino.tests.product.launcher.env.DockerContainer#execCommand` signature is wrong), as it suggests totally different semantics

let's switch to using a single String instead.
that would also let us realize we should quote the `path` within shell command

**Comment by kokosing** [comment]
```
iner 'hadoop-master' path /var/log/container-health.log
2021-04-13T08:50:06.8822823Z java.lang.RuntimeException: Could not execute command 'test -d /var/log/container-health.log && /usr/bin/find /var/log/container-health.log -type f -print' in container hadoop-master: test: extra argument ‘&&’
2021-04-13T08:50:06.8824077Z 
2021-04-13T08:50:06.8825640Z 	at io.trino.tests.product.launcher.env.DockerContainer.execCommand(DockerContainer.java:255)
2021-04-13T08:50:06.8828926Z 	at io.trino.tests.product.launcher.env.DockerContainer.listFilesInContainer(DockerContainer.java:353)
2021-04-13T08:50:06.8832384Z 	at io.trino.tests.product.launcher.env.DockerContainer.copyLogsToHostPath(DockerContainer.java:282)
2021-04-13T08:50:06.8835389Z 	at io.trino.tests.product.launcher.env.EnvironmentListener$3.containerStopping(EnvironmentListener.java:212)
2021-04-13T08:50:06.8838057Z 	at io.trino.tests.product.launcher.env.EnvironmentListener$1.lambda$containerStopping$6(EnvironmentListener.java:139)
2021-04-13T08:50:06.8840970Z 	at io.trino.tests.product.launcher.env.EnvironmentListener.lambda$tryInvokeListener$0(EnvironmentListener.java:80)
2021-04-13T08:50:06.8843215Z 	at net.jodah.failsafe.Functions.lambda$toCtxSupplier$13(Functions.java:261)
2021-04-13T08:50:06.8844613Z 	at net.jodah.failsafe.Functions.lambda$getPromise$1(Functions.java:81)
2021-04-13T08:50:06.8845920Z 	at net.jodah.failsafe.Functions.lambda$null$3(Functions.java:103)
2021-04-13T08:50:06.8847942Z 	at net.jodah.failsafe.internal.util.DelegatingScheduler.lambda$schedule$0(DelegatingScheduler.java:141)
2021-04-13T08:50:06.8850214Z 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2021-04-13T08:50:06.8852013Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
2021-04-13T08:50:06.8854017Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
2021-04-13T08:50:06.8855336Z 	at java.base/java.lang.Thread.run(Thread.java:834)
2021-04-13T08:50:06.8855814Z 
```

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/DockerContainer.java (hunk 1 above)
Comment (findepi) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/DockerContainer.java:353]: per https://github.com/trinodb/trino/pull/7578#issuecomment-818572530 ... it's even worse than that.

**Comment by kokosing** [comment]
Yet another try.

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/DockerContainer.java [hunk 2]
```diff
@@ -348,9 +353,14 @@ private void copyFileFromContainer(String filename, Path targetPath)
     private List<String> listFilesInContainer(String path)
     {
         try {
+            ExecResult execResult = execCommandForResult("/usr/bin/find", path, "-type", "f", "-print");
+            if (execResult.getExitCode() != 0) {
+                log.warn("Unable to list files from '%s' in container %s: %s", path, logicalName, execResult.getStderr());
```
Comment (losipiuk) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/DockerContainer.java:358]: nit: use same wording as in `catch` clause

**Review by losipiuk — APPROVED** [review]
Seems fine

**Review by wendigo — APPROVED** [review]

**Comment by kokosing** [comment]
CI hit: https://github.com/trinodb/trino/issues/7534

**Pull request merged**